sys-kernel/gentoo-kernel/gentoo-kernel-6.3.8-r1.ebuild

```diff
@@ -136,5 +136,48 @@ src_prepare() {
 		merge_configs+=( "${dist_conf_path}/big-endian.config" )
 	fi
 
+	if use modules-sign; then
+		: "${MODULES_SIGN_HASH:=sha512}"
```

Comment:
Please move the core of this logic into `kernel-build`, and document `MODULES_SIGN_HASH` and `MODULES_SIGN_KEY` as `@USER_VARIABLE`.

Maybe add a pre-inherit var that enables module signing, i.e. adds `IUSE`, modifies `kernel-build_merge_configs` and adds postinst.

Reply:
The reason I put the part that manipulates the CONFIG_* in the ebuild instead of the eclass is because if the way this is configured were to change in the future we would have to change the eclass again and put a bunch of extra conditions in it. Plus all the other CONFIG_* stuff is defined in the ebuild.

If we can be reasonably sure that the CONFIG_MODULE_SIG_* stuff won't change then I can move all this into the eclass, no problem.

Reply:
It's fine, we can either use conditions on `PV` or have the variable specify "ABI version".